doc/api/pyplot_summary.rst

```diff
@@ -1,29 +1,283 @@
-Pyplot function overview
-------------------------
+******
+pyplot
+******
 
-.. currentmodule:: matplotlib
+.. currentmodule:: matplotlib.pyplot
+
+.. automodule:: matplotlib.pyplot
+    :no-members:
+
+Information
+===========
 
 .. autosummary::
-   :toctree: _as_gen
-   :template: autofunctions.rst
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
 
-   pyplot
+    colormaps
+    get_plot_commands
 
+Global functions
+================
 
-.. currentmodule:: matplotlib.pyplot
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
 
-.. autofunction:: plotting
+    findobj
+    install_repl_displayhook
+    ioff
+    ion
+    isinteractive
+    pause
+    rc
+    rc_context
+    rcdefaults
+    show
+    switch_backend
+    uninstall_repl_displayhook
 
+Any artist
+==========
 
-Colors in Matplotlib
---------------------
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
 
-There are many colormaps you can use to map data onto color values.
-Below we list several ways in which color can be utilized in Matplotlib.
+    getp
+    setp
+    xkcd
 
-For a more in-depth look at colormaps, see the
-:doc:`/tutorials/colors/colormaps` tutorial.
+Making figures
+==============
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
 
-.. currentmodule:: matplotlib.pyplot
+    figure
+    subplots
+
+Working with figures
+====================
+
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
+
+    clf
+    close
+    connect
+    disconnect
+    draw
+    fignum_exists
+    figure
+    gcf
+    get_current_fig_manager
+    get_figlabels
+    get_fignums
+    ginput
+    savefig
+    waitforbuttonpress
+
+Putting things in figures
+=========================
+
+.. autosummary::
+    :toctree: _as_gen
+    :template: autosummary.rst
+    :nosignatures:
+
+    figimage
+    figlegend
+    figtext
+    suptitle
+
+Working with axes
+=================
+
```

Comment:
Might be worth putting a single sentence here saying what an `axes` is (as opposed to an `axis`)